components/user/UserProfile.js

```diff
@@ -33,6 +34,10 @@ function UserProfile({ BASE_URL, data }) {
     </Link>
   );
 
+  if (!data) {
+    return <p className="text-3xl text-primary-low">No profile data</p>;
+  }
```

Comment:
This check should be done outside the component, so the component is not rendered

Reply:
Done!